cfn-resources/project-invitation/cmd/resource/resource.go

```diff
@@ -34,237 +30,59 @@ var UpdateRequiredFields = []string{constants.ProjectID, constants.ID}
-	if err != nil {
-		_, _ = log.Warnf("Delete - error: %+v", err)
-		return progressevents.GetFailedEventByResponse(err.Error(), res.Response), nil
-	}
-	_, _ = log.Debugf("deleted invitation with Id :%s", *currentModel.Id)
-
-	// Response
-	return handler.ProgressEvent{
-		OperationStatus: handler.Success,
-		ResourceModel:   nil,
-	}, nil
+	return DeleteOp(req, prevModel, currentModel)
 }
 
 func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
-	setup()
-
-	_, _ = log.Debugf("List() currentModel:%+v", currentModel)
-
-	// Validation
-	modelValidation := validateModel(ListRequiredFields, currentModel)
-	if modelValidation != nil {
-		return *modelValidation, nil
-	}
+	return ListOp(req, prevModel, currentModel)
+}
 
-	// Create atlas client
-	if currentModel.Profile == nil || *currentModel.Profile == "" {
-		currentModel.Profile = aws.String(profile.DefaultProfile)
-	}
+func validateModel(fields []string, model *Model) *handler.ProgressEvent {
+	return validator.ValidateModel(fields, model)
+}
 
-	client, peErr := util.NewMongoDBClient(req, currentModel.Profile)
-	if peErr != nil {
-		return *peErr, nil
-	}
+func setup() {
+	util.SetupLogger("mongodb-atlas-project-invitation")
+}
 
-	listOptions := &mongodbatlas.InvitationOptions{
-		Username: *currentModel.Username,
-	}
-	invitations, res, err := client.Projects.Invitations(context.Background(), *currentModel.ProjectId, listOptions)
+func validateProjectInvitationAlreadyAccepted(ctx context.Context, client *util.MongoDBClient, username, projectID string) (bool, error) {
+	user, _, err := client.Atlas.AtlasUsers.GetByName(ctx, username)
 	if err != nil {
-		_, _ = log.Warnf("List - error: %+v", err)
-		return progressevents.GetFailedEventByResponse(err.Error(), res.Response), nil
+		return false, err
 	}
-
-	var invites []interface{}
-	// iterate invites
-	for i := range invitations {
-		invite := invitationToModel(currentModel, invitations[i])
-		invites = append(invites, invite)
+	for _, role := range user.Roles {
+		if role.GroupID == projectID {
+			return true, nil
+		}
 	}
 
-	// Response
-	return handler.ProgressEvent{
-		OperationStatus: handler.Success,
-		ResourceModels:  invites,
-	}, nil
+	return false, nil
 }
 
-func invitationToModel(currentModel *Model, invitation *mongodbatlas.Invitation) Model {
+func invitationAPIRequestToModel(currentModel *Model, invitation *admin.GroupInvitation) Model {
 	out := Model{
 		Profile:         currentModel.Profile,
 		ProjectId:       currentModel.ProjectId,
-		Username:        &invitation.Username,
-		Id:              &invitation.ID,
+		Username:        invitation.Username,
+		Id:              invitation.Id,
 		Roles:           invitation.Roles,
-		ExpiresAt:       &invitation.ExpiresAt,
-		CreatedAt:       &invitation.CreatedAt,
-		InviterUsername: &invitation.InviterUsername,
+		ExpiresAt:       util.StringPointerIfNotEmpty(invitation.ExpiresAt.GoString()),
```

Comment:
why GoString and StringPointerIfNotEmpty?

Reply:
`invitation.ExpiresAt` is of type `*time.Time` but the model wants a `*string`

Reply:
but I'm not sure that we're using ISO here, I think we should depend on util.TimeToString here, validating first if time ptr is not nil

Reply:
Thanks! I didn't see TimeToString. Updated